### ISSUE TITLE
tsm1 engine - panic: runtime error: index out of range

### ISSUE THREAD

**Issue by victorhooi** [influxdb/influxdb#4419]
I am running InfluxDB 0.9.5-nightly-f1e0c59

I ran some queries from Grafana, which seems to lock up the box (basically, Grafana became unresponsive, and influxdb was using about 2000% CPU).

I ran `sudo service influxdb stop` to stop the service (which I assumed would do a clean shutdown), and then tried to start it again.

However, influxdb now crashes every time I attempt to start it.

Here is the relevant part of the logfile:
```
Oct 13 01:49:00 ip-172-31-9-47 influxd[1956]: [query] 2015/10/13 01:49:00 SELECT mean(duration_in_milliseconds) AS "duration_in_milliseconds" FROM "insight"."default".operations WHERE project = 'CS-23893' AND hostname = 'slcdbx1399.slc.ebay.com' AND time > 1444410203s AND time < 1444413871s GROUP BY time(1s)
Oct 13 01:50:46 ip-172-31-9-47 systemd[1]: Stopping InfluxDB is an open-source, distributed, time series database...
Oct 13 01:50:46 ip-172-31-9-47 influxd[1956]: [run] 2015/10/13 01:50:46 Signal received, initializing clean shutdown...
Oct 13 01:50:46 ip-172-31-9-47 influxd[1956]: [run] 2015/10/13 01:50:46 Waiting for clean shutdown...
Oct 13 01:50:46 ip-172-31-9-47 influxd[1956]: [tcp] 2015/10/13 01:50:46 cluster service accept error: network connection closed
Oct 13 01:50:46 ip-172-31-9-47 influxd[1956]: [shard-precreation] 2015/10/13 01:50:46 Precreation service terminating
Oct 13 01:50:46 ip-172-31-9-47 influxd[1956]: [snapshot] 2015/10/13 01:50:46 snapshot listener closed
Oct 13 01:50:46 ip-172-31-9-47 influxd[1956]: [copier] 2015/10/13 01:50:46 copier listener closed
Oct 13 01:50:46 ip-172-31-9-47 influxd[1956]: [continuous_querier] 2015/10/13 01:50:46 continuous query service terminating
Oct 13 01:50:46 ip-172-31-9-47 influxd[1956]: [retention] 2015/10/13 01:50:46 retention policy enforcement terminating
Oct 13 01:50:46 ip-172-31-9-47 influxd[1956]: [monitor] 2015/10/13 01:50:46 shutting down monitor system
Oct 13 01:50:46 ip-172-31-9-47 influxd[1956]: [monitor] 2015/10/13 01:50:46 terminating storage of statistics
Oct 13 01:51:16 ip-172-31-9-47 systemd[1]: Stopped InfluxDB is an open-source, distributed, time series database.
Oct 13 01:55:08 ip-172-31-9-47 systemd[1]: Started InfluxDB is an open-source, distributed, time series database.
Oct 13 01:55:08 ip-172-31-9-47 systemd[1]: Starting InfluxDB is an open-source, distributed, time series database...
Oct 13 01:55:08 ip-172-31-9-47 influxd[35973]: 8888888           .d888 888                   8888888b.  888888b.
Oct 13 01:55:08 ip-172-31-9-47 influxd[35973]: 888            d88P"  888                   888  "Y88b 888  "88b
Oct 13 01:55:08 ip-172-31-9-47 influxd[35973]: 888            888    888                   888    888 888  .88P
Oct 13 01:55:08 ip-172-31-9-47 influxd[35973]: 888   88888b.  888888 888 888  888 888  888 888    888 8888888K.
Oct 13 01:55:08 ip-172-31-9-47 influxd[35973]: 888   888 "88b 888    888 888  888  Y8bd8P' 888    888 888  "Y88b
Oct 13 01:55:08 ip-172-31-9-47 influxd[35973]: 888   888  888 888    888 888  888   X88K   888    888 888    888
Oct 13 01:55:08 ip-172-31-9-47 influxd[35973]: 888   888  888 888    888 Y88b 888 .d8""8b. 888  .d88P 888   d88P
Oct 13 01:55:08 ip-172-31-9-47 influxd[35973]: 8888888 888  888 888    888  "Y88888 888  888 8888888P"  8888888P"
Oct 13 01:55:08 ip-172-31-9-47 influxd[35973]: 2015/10/13 01:55:08 InfluxDB starting, version 0.9.5-nightly-f1e0c59, branch master, commit f1e0c5938f4cbb009e4b5ba3160a1bd281c2f003, built '2015-10-11T04:00:41+0000'
Oct 13 01:55:08 ip-172-31-9-47 influxd[35973]: 2015/10/13 01:55:08 Go version go1.5.1, GOMAXPROCS set to 36
Oct 13 01:55:08 ip-172-31-9-47 influxd[35973]: 2015/10/13 01:55:08 Using configuration at: /etc/opt/influxdb/influxdb.conf
Oct 13 01:55:08 ip-172-31-9-47 influxd[35973]: [metastore] 2015/10/13 01:55:08 Using data dir: /var/opt/influxdb/meta
Oct 13 01:55:08 ip-172-31-9-47 influxd[35973]: [metastore] 2015/10/13 01:55:08 Skipping cluster join: already member of cluster: nodeId=1 raftEnabled=true peers=[localhost:8088]
Oct 13 01:55:08 ip-172-31-9-47 influxd[35973]: [metastore] 2015/10/13 01:55:08 Node at localhost:8088 [Follower]
Oct 13 01:55:09 ip-172-31-9-47 influxd[35973]: [metastore] 2015/10/13 01:55:09 Node at localhost:8088 [Leader]. peers=[localhost:8088]
Oct 13 01:55:09 ip-172-31-9-47 influxd[35973]: [monitor] 2015/10/13 01:55:09 Starting monitor system
Oct 13 01:55:09 ip-172-31-9-47 influxd[35973]: [monitor] 2015/10/13 01:55:09 'build' registered for diagnostics monitoring
Oct 13 01:55:09 ip-172-31-9-47 influxd[35973]: [monitor] 2015/10/13 01:55:09 'runtime' registered for diagnostics monitoring
Oct 13 01:55:09 ip-172-31-9-47 influxd[35973]: [monitor] 2015/10/13 01:55:09 'network' registered for diagnostics monitoring
Oct 13 01:55:09 ip-172-31-9-47 influxd[35973]: [monitor] 2015/10/13 01:55:09 'system' registered for diagnostics monitoring
Oct 13 01:55:09 ip-172-31-9-47 influxd[35973]: [store] 2015/10/13 01:55:09 Using data dir: /var/opt/influxdb/data
Oct 13 01:55:09 ip-172-31-9-47 influxd[35973]: [tsm1wal] 2015/10/13 01:55:09 tsm1 WAL starting with 5242880 flush memory size threshold and 104857600 max memory size threshold
Oct 13 01:55:09 ip-172-31-9-47 influxd[35973]: [monitor] 2015/10/13 01:55:09 Storing statistics in database '_internal' retention policy '', at interval 10s
Oct 13 01:55:09 ip-172-31-9-47 influxd[35973]: [tsm1wal] 2015/10/13 01:55:09 tsm1 WAL writing to /var/opt/influxdb/data/_internal/monitor/1
Oct 13 01:55:09 ip-172-31-9-47 influxd[35973]: [tsm1wal] 2015/10/13 01:55:09 startup flush of /var/opt/influxdb/data/_internal/monitor/1 with 0 keys and 0 total values of 0 bytes
Oct 13 01:55:09 ip-172-31-9-47 influxd[35973]: [tsm1wal] 2015/10/13 01:55:09 /var/opt/influxdb/data/_internal/monitor/1 flush to index took 2.44µs
Oct 13 01:55:09 ip-172-31-9-47 influxd[35973]: [tsm1wal] 2015/10/13 01:55:09 tsm1 WAL starting with 5242880 flush memory size threshold and 104857600 max memory size threshold
Oct 13 01:55:09 ip-172-31-9-47 influxd[35973]: [tsm1wal] 2015/10/13 01:55:09 tsm1 WAL writing to /var/opt/influxdb/data/_internal/monitor/11
Oct 13 01:55:10 ip-172-31-9-47 influxd[35973]: [tsm1wal] 2015/10/13 01:55:10 startup flush of /var/opt/influxdb/data/_internal/monitor/11 with 47 keys and 31251 total values of 500016 bytes
Oct 13 01:55:10 ip-172-31-9-47 influxd[35973]: [tsm1] 2015/10/13 01:55:10 writing new index file /var/opt/influxdb/data/_internal/monitor/11/0000001.tsm1
Oct 13 01:55:10 ip-172-31-9-47 influxd[35973]: [tsm1wal] 2015/10/13 01:55:10 /var/opt/influxdb/data/_internal/monitor/11 flush to index took 23.873601ms
Oct 13 01:55:10 ip-172-31-9-47 influxd[35973]: [tsm1wal] 2015/10/13 01:55:10 tsm1 WAL starting with 5242880 flush memory size threshold and 104857600 max memory size threshold
Oct 13 01:55:10 ip-172-31-9-47 influxd[35973]: [tsm1wal] 2015/10/13 01:55:10 tsm1 WAL writing to /var/opt/influxdb/data/_internal/monitor/4
Oct 13 01:55:10 ip-172-31-9-47 influxd[35973]: [tsm1wal] 2015/10/13 01:55:10 startup flush of /var/opt/influxdb/data/_internal/monitor/4 with 0 keys and 0 total values of 0 bytes
Oct 13 01:55:10 ip-172-31-9-47 influxd[35973]: [tsm1wal] 2015/10/13 01:55:10 /var/opt/influxdb/data/_internal/monitor/4 flush to index took 1.6µs
Oct 13 01:55:10 ip-172-31-9-47 influxd[35973]: [tsm1wal] 2015/10/13 01:55:10 tsm1 WAL starting with 5242880 flush memory size threshold and 104857600 max memory size threshold
Oct 13 01:55:10 ip-172-31-9-47 influxd[35973]: [tsm1wal] 2015/10/13 01:55:10 tsm1 WAL writing to /var/opt/influxdb/data/_internal/monitor/6
Oct 13 01:55:10 ip-172-31-9-47 influxd[35973]: [tsm1wal] 2015/10/13 01:55:10 startup flush of /var/opt/influxdb/data/_internal/monitor/6 with 0 keys and 0 total values of 0 bytes
Oct 13 01:55:10 ip-172-31-9-47 influxd[35973]: [tsm1wal] 2015/10/13 01:55:10 /var/opt/influxdb/data/_internal/monitor/6 flush to index took 958ns
Oct 13 01:55:10 ip-172-31-9-47 influxd[35973]: [tsm1wal] 2015/10/13 01:55:10 tsm1 WAL starting with 5242880 flush memory size threshold and 104857600 max memory size threshold
Oct 13 01:55:10 ip-172-31-9-47 influxd[35973]: [tsm1wal] 2015/10/13 01:55:10 tsm1 WAL writing to /var/opt/influxdb/data/_internal/monitor/7
Oct 13 01:55:10 ip-172-31-9-47 influxd[35973]: [tsm1wal] 2015/10/13 01:55:10 startup flush of /var/opt/influxdb/data/_internal/monitor/7 with 55 keys and 330 total values of 5280 bytes
Oct 13 01:55:10 ip-172-31-9-47 influxd[35973]: [tsm1] 2015/10/13 01:55:10 rewriting index file /var/opt/influxdb/data/_internal/monitor/7/0000765.tsm1 with /var/opt/influxdb/data/_internal/monitor/7/0000767.tsm1
Oct 13 01:55:10 ip-172-31-9-47 influxd[35973]: panic: runtime error: index out of range
Oct 13 01:55:10 ip-172-31-9-47 influxd[35973]: goroutine 98 [running]:
Oct 13 01:55:10 ip-172-31-9-47 influxd[35973]: github.com/influxdb/influxdb/tsdb/engine/tsm1.Values.MinTime(0xc820213de0, 0x0, 0x2, 0x140bf8bbb146cbdb)
Oct 13 01:55:10 ip-172-31-9-47 influxd[35973]: /tmp/tmp.GEQALQdekA/src/github.com/influxdb/influxdb/tsdb/engine/tsm1/encoding.go:65 +0x75
Oct 13 01:55:10 ip-172-31-9-47 influxd[35973]: github.com/influxdb/influxdb/tsdb/engine/tsm1.(*Engine).DecodeAndCombine(0xc8201b5560, 0xc820213de0, 0x0, 0x2, 0x7f088808ddc8, 0xbebe, 0x492988, 0xc820315000, 0x0, 0x4e20, ...)
Oct 13 01:55:10 ip-172-31-9-47 influxd[35973]: /tmp/tmp.GEQALQdekA/src/github.com/influxdb/influxdb/tsdb/engine/tsm1/tsm1.go:1668 +0x6bd
Oct 13 01:55:10 ip-172-31-9-47 influxd[35973]: github.com/influxdb/influxdb/tsdb/engine/tsm1.(*Engine).rewriteFile(0xc8201b5560, 0xc82000ca20, 0xc820235b60, 0x0, 0x0)
Oct 13 01:55:10 ip-172-31-9-47 influxd[35973]: /tmp/tmp.GEQALQdekA/src/github.com/influxdb/influxdb/tsdb/engine/tsm1/tsm1.go:1111 +0x208f
Oct 13 01:55:10 ip-172-31-9-47 influxd[35973]: github.com/influxdb/influxdb/tsdb/engine/tsm1.(*Engine).Write.func1(0xc8201b5560, 0xc82029d100, 0xc8205bdfa0, 0xc82029d110, 0xc82000ca20, 0xc820235b60)
Oct 13 01:55:10 ip-172-31-9-47 influxd[35973]: /tmp/tmp.GEQALQdekA/src/github.com/influxdb/influxdb/tsdb/engine/tsm1/tsm1.go:392 +0x3f
Oct 13 01:55:10 ip-172-31-9-47 influxd[35973]: created by github.com/influxdb/influxdb/tsdb/engine/tsm1.(*Engine).Write
Oct 13 01:55:10 ip-172-31-9-47 influxd[35973]: /tmp/tmp.GEQALQdekA/src/github.com/influxdb/influxdb/tsdb/engine/tsm1/tsm1.go:398 +0x69b
Oct 13 01:55:10 ip-172-31-9-47 influxd[35973]: goroutine 1 [semacquire]:
Oct 13 01:55:10 ip-172-31-9-47 influxd[35973]: sync.runtime_Semacquire(0xc82029d11c)
Oct 13 01:55:10 ip-172-31-9-47 influxd[35973]: /root/.gvm/gos/go1.5.1/src/runtime/sema.go:43 +0x26
Oct 13 01:55:10 ip-172-31-9-47 influxd[35973]: sync.(*WaitGroup).Wait(0xc82029d110)
Oct 13 01:55:10 ip-172-31-9-47 influxd[35973]: /root/.gvm/gos/go1.5.1/src/sync/waitgroup.go:126 +0xb4
Oct 13 01:55:10 ip-172-31-9-47 influxd[35973]: github.com/influxdb/influxdb/tsdb/engine/tsm1.(*Engine).Write(0xc8201b5560, 0xc820235980, 0xc820234bd0, 0x0, 0x0, 0x0, 0x0, 0x0)
Oct 13 01:55:10 ip-172-31-9-47 influxd[35973]: /tmp/tmp.GEQALQdekA/src/github.com/influxdb/influxdb/tsdb/engine/tsm1/tsm1.go:436 +0x731
Oct 13 01:55:10 ip-172-31-9-47 influxd[35973]: github.com/influxdb/influxdb/tsdb/engine/tsm1.(*Log).flush(0xc8201b5440, 0x4, 0x0, 0x0)
Oct 13 01:55:10 ip-172-31-9-47 systemd[1]: influxdb.service: main process exited, code=exited, status=2/INVALIDARGUMENT
Oct 13 01:55:10 ip-172-31-9-47 systemd[1]: Unit influxdb.service entered failed state.
Oct 13 01:55:10 ip-172-31-9-47 systemd[1]: influxdb.service failed.
Oct 13 01:55:10 ip-172-31-9-47 systemd[1]: influxdb.service holdoff time over, scheduling restart.
```

Just in case, this is the last 1000 lines of the logfile, covering from when I stopped it and tried to start it again:

https://gist.github.com/victorhooi/1c242d9bda9025a6c8d0

It seems I'm going to have to blow away all my data and start again =(. However, happy to keep them around if they would help you guys debug?

**Comment by jmcook** [comment]
Getting the same crash with the same build.

**Comment by huhongbo** [comment]
I have the same problem
[tsm1] 2015/10/16 13:50:53 rewriting index file /var/opt/influxdb/data/metrics/default/14/0023089.tsm1 with /var/opt/influxdb/data/metrics/default/14/0023119.tsm1
panic: runtime error: index out of range

**Comment by jwilder** [comment]
This might be fixed by #4402

**Comment by beckettsean** [comment]
closing as duplicate of https://github.com/influxdb/influxdb/issues/4365
